commons/src/main/java/fr/free/nrw/commons/upload/ShareActivity.java

```diff
@@ -175,13 +175,24 @@ public void onCreate(Bundle savedInstanceState) {
         mediaUriString = mediaUri.toString();
         Log.d("Image", "Uri: " + mediaUriString);
 
+        //convert image Uri to file path
         FilePathConverter uriObj = new FilePathConverter(this, mediaUri);
         String filePath = uriObj.getFilePath();
 
+        //extract the coordinates of image in decimal degrees
         GPSExtractor imageObj = new GPSExtractor(filePath);
         String coords = imageObj.getCoords();
         Log.d("Image", "Coords of image: " + coords);
 
+        //build URL with image coords for MediaWiki API calls
+        String apiUrl = UrlBuilder.buildUrl(coords);
+        Log.d("Image", "URL: " + apiUrl);
+
+
+        //asynchronous calls to MediaWiki Commons API to match image coords with nearby Commons categories
+        MwVolleyApi apiCall = new MwVolleyApi(this);
```

Comment:
Although it's possible to new a MwVolleyApi each time, it feels like it might be more natural to have the same lifetime as the Activity.